src/Symfony/Component/Cache/Adapter/FilesystemAdapter.php

```diff
@@ -147,7 +147,7 @@ protected function doSave(array $values, $lifetime)
     private function getFile($id, $mkdir = false)
     {
         $hash = str_replace('/', '-', base64_encode(md5($id, true)));
```

Comment:
> As far as security is concerned, if a collision occurs, it will be caught by the unhashed key comparison done when opening files.

This doesn't solve the security problem. When a cache key doesn't exist which it looks like when a collision happens with the key comparison, then clients usually write the cache with the collided key. So it will overwrite the cache of another key. This basically means anybody can prune the cache by using collisions.